diff -r -u -N -w Main\include\auto\Mogre-WIN32_OgreTimerImp.h Main\include\auto\Mogre-WIN32_OgreTimerImp.h
--- Main\include\auto\Mogre-WIN32_OgreTimerImp.h	2013-04-22 20:21:41.967636200 +0200
+++ Main\include\auto\Mogre-WIN32_OgreTimerImp.h	2013-04-22 20:23:13.868644100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "WIN32/OgreTimerImp.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreAnimable.h Main\include\auto\MogreAnimable.h
--- Main\include\auto\MogreAnimable.h	2013-04-22 20:21:37.191832800 +0200
+++ Main\include\auto\MogreAnimable.h	2013-04-22 20:23:13.881653000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreAnimable.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreStringVector.h"
 
diff -r -u -N -w Main\include\auto\MogreAnimation.h Main\include\auto\MogreAnimation.h
--- Main\include\auto\MogreAnimation.h	2013-04-22 20:21:37.289900000 +0200
+++ Main\include\auto\MogreAnimation.h	2013-04-22 20:23:13.889660400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreAnimation.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreAnimationTrack.h"
 #include "MogreAnimationState.h"
diff -r -u -N -w Main\include\auto\MogreAnimationState.h Main\include\auto\MogreAnimationState.h
--- Main\include\auto\MogreAnimationState.h	2013-04-22 20:21:37.342936900 +0200
+++ Main\include\auto\MogreAnimationState.h	2013-04-22 20:23:13.899666300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreAnimationState.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreAnimationTrack.h Main\include\auto\MogreAnimationTrack.h
--- Main\include\auto\MogreAnimationTrack.h	2013-04-22 20:21:37.409480600 +0200
+++ Main\include\auto\MogreAnimationTrack.h	2013-04-22 20:23:13.939667600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreAnimationTrack.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePose.h"
 
diff -r -u -N -w Main\include\auto\MogreArchive.h Main\include\auto\MogreArchive.h
--- Main\include\auto\MogreArchive.h	2013-04-22 20:21:37.434499600 +0200
+++ Main\include\auto\MogreArchive.h	2013-04-22 20:23:13.946672100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreArchive.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreArchiveManager.h Main\include\auto\MogreArchiveManager.h
--- Main\include\auto\MogreArchiveManager.h	2013-04-22 20:21:37.451509500 +0200
+++ Main\include\auto\MogreArchiveManager.h	2013-04-22 20:23:13.951678100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreArchiveManager.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreBillboard.h Main\include\auto\MogreBillboard.h
--- Main\include\auto\MogreBillboard.h	2013-04-22 20:21:37.476527700 +0200
+++ Main\include\auto\MogreBillboard.h	2013-04-22 20:23:13.956681200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreBillboard.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
 
diff -r -u -N -w Main\include\auto\MogreBillboardChain.h Main\include\auto\MogreBillboardChain.h
--- Main\include\auto\MogreBillboardChain.h	2013-04-22 20:21:37.563583600 +0200
+++ Main\include\auto\MogreBillboardChain.h	2013-04-22 20:23:13.966688300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreBillboardChain.h"
+#pragma managed(pop)
 #include "MogreRenderable.h"
 #include "MogreMovableObject.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreBillboardParticleRenderer.h Main\include\auto\MogreBillboardParticleRenderer.h
--- Main\include\auto\MogreBillboardParticleRenderer.h	2013-04-22 20:21:37.591603200 +0200
+++ Main\include\auto\MogreBillboardParticleRenderer.h	2013-04-22 20:23:13.972690700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreBillboardParticleRenderer.h"
+#pragma managed(pop)
 #include "MogreParticleSystemRenderer.h"
 #include "MogreBillboardSet.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreBillboardSet.h Main\include\auto\MogreBillboardSet.h
--- Main\include\auto\MogreBillboardSet.h	2013-04-22 20:21:37.676652400 +0200
+++ Main\include\auto\MogreBillboardSet.h	2013-04-22 20:23:13.981695700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreBillboardSet.h"
+#pragma managed(pop)
 #include "MogreRenderable.h"
 #include "MogreMovableObject.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreBlendMode.h Main\include\auto\MogreBlendMode.h
--- Main\include\auto\MogreBlendMode.h	2013-04-22 20:21:37.706157300 +0200
+++ Main\include\auto\MogreBlendMode.h	2013-04-22 20:23:13.986702900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreBlendMode.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreBone.h Main\include\auto\MogreBone.h
--- Main\include\auto\MogreBone.h	2013-04-22 20:21:37.736924900 +0200
+++ Main\include\auto\MogreBone.h	2013-04-22 20:23:13.992388300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreBone.h"
+#pragma managed(pop)
 #include "MogreNode.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreBorderPanelOverlayElement.h Main\include\auto\MogreBorderPanelOverlayElement.h
--- Main\include\auto\MogreBorderPanelOverlayElement.h	2013-04-22 20:21:37.810976400 +0200
+++ Main\include\auto\MogreBorderPanelOverlayElement.h	2013-04-22 20:23:13.997392700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreBorderPanelOverlayElement.h"
+#pragma managed(pop)
 #include "MogrePanelOverlayElement.h"
 #include "MogrePrerequisites.h"
 #include "MogreOverlayElement.h"
diff -r -u -N -w Main\include\auto\MogreCamera.h Main\include\auto\MogreCamera.h
--- Main\include\auto\MogreCamera.h	2013-04-22 20:21:37.926051500 +0200
+++ Main\include\auto\MogreCamera.h	2013-04-22 20:23:14.005393600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreCamera.h"
+#pragma managed(pop)
 #include "MogreFrustum.h"
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
diff -r -u -N -w Main\include\auto\MogreCodec.h Main\include\auto\MogreCodec.h
--- Main\include\auto\MogreCodec.h	2013-04-22 20:21:37.946066900 +0200
+++ Main\include\auto\MogreCodec.h	2013-04-22 20:23:14.012400500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreCodec.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreStringVector.h"
 
diff -r -u -N -w Main\include\auto\MogreColourValue.h Main\include\auto\MogreColourValue.h
--- Main\include\auto\MogreColourValue.h	2013-04-22 20:21:37.955075200 +0200
+++ Main\include\auto\MogreColourValue.h	2013-04-22 20:23:14.053089400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreColourValue.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreCommon.h Main\include\auto\MogreCommon.h
--- Main\include\auto\MogreCommon.h	2013-04-22 20:21:37.961079200 +0200
+++ Main\include\auto\MogreCommon.h	2013-04-22 20:23:14.063096500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreCommon.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreCompositionPass.h Main\include\auto\MogreCompositionPass.h
--- Main\include\auto\MogreCompositionPass.h	2013-04-22 20:21:37.985466800 +0200
+++ Main\include\auto\MogreCompositionPass.h	2013-04-22 20:23:14.073103600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreCompositionPass.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
 #include "MogreRenderSystem.h"
diff -r -u -N -w Main\include\auto\MogreCompositionTargetPass.h Main\include\auto\MogreCompositionTargetPass.h
--- Main\include\auto\MogreCompositionTargetPass.h	2013-04-22 20:21:37.998478200 +0200
+++ Main\include\auto\MogreCompositionTargetPass.h	2013-04-22 20:23:14.081108200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreCompositionTargetPass.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
 
diff -r -u -N -w Main\include\auto\MogreCompositionTechnique.h Main\include\auto\MogreCompositionTechnique.h
--- Main\include\auto\MogreCompositionTechnique.h	2013-04-22 20:21:38.019490400 +0200
+++ Main\include\auto\MogreCompositionTechnique.h	2013-04-22 20:23:14.087113000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreCompositionTechnique.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePixelFormat.h"
 
diff -r -u -N -w Main\include\auto\MogreCompositor.h Main\include\auto\MogreCompositor.h
--- Main\include\auto\MogreCompositor.h	2013-04-22 20:21:38.048509100 +0200
+++ Main\include\auto\MogreCompositor.h	2013-04-22 20:23:14.096120400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreCompositor.h"
+#pragma managed(pop)
 #include "MogreResource.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreCompositorChain.h Main\include\auto\MogreCompositorChain.h
--- Main\include\auto\MogreCompositorChain.h	2013-04-22 20:21:38.066452200 +0200
+++ Main\include\auto\MogreCompositorChain.h	2013-04-22 20:23:14.102125300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreCompositorChain.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreRenderTargetListener.h"
 
diff -r -u -N -w Main\include\auto\MogreCompositorInstance.h Main\include\auto\MogreCompositorInstance.h
--- Main\include\auto\MogreCompositorInstance.h	2013-04-22 20:21:38.101475400 +0200
+++ Main\include\auto\MogreCompositorInstance.h	2013-04-22 20:23:14.110127800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreCompositorInstance.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreCompositorManager.h Main\include\auto\MogreCompositorManager.h
--- Main\include\auto\MogreCompositorManager.h	2013-04-22 20:21:38.146505700 +0200
+++ Main\include\auto\MogreCompositorManager.h	2013-04-22 20:23:14.117136700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreCompositorManager.h"
+#pragma managed(pop)
 #include "MogreResourceManager.h"
 #include "MogrePrerequisites.h"
 #include "MogrePixelFormat.h"
diff -r -u -N -w Main\include\auto\MogreConfigFile.h Main\include\auto\MogreConfigFile.h
--- Main\include\auto\MogreConfigFile.h	2013-04-22 20:21:38.172529700 +0200
+++ Main\include\auto\MogreConfigFile.h	2013-04-22 20:23:14.122137000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreConfigFile.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreStringVector.h"
 
diff -r -u -N -w Main\include\auto\MogreConfigOptionMap.h Main\include\auto\MogreConfigOptionMap.h
--- Main\include\auto\MogreConfigOptionMap.h	2013-04-22 20:21:38.186534100 +0200
+++ Main\include\auto\MogreConfigOptionMap.h	2013-04-22 20:23:14.131141600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreConfigOptionMap.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreStringVector.h"
 
diff -r -u -N -w Main\include\auto\MogreDataStream.h Main\include\auto\MogreDataStream.h
--- Main\include\auto\MogreDataStream.h	2013-04-22 20:21:38.211550200 +0200
+++ Main\include\auto\MogreDataStream.h	2013-04-22 20:23:14.138146100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreDataStream.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
 
diff -r -u -N -w Main\include\auto\MogreDDSCodec.h Main\include\auto\MogreDDSCodec.h
--- Main\include\auto\MogreDDSCodec.h	2013-04-22 20:21:38.225557100 +0200
+++ Main\include\auto\MogreDDSCodec.h	2013-04-22 20:23:14.145151400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreDDSCodec.h"
+#pragma managed(pop)
 #include "MogreImageCodec.h"
 #include "MogrePrerequisites.h"
 #include "MogreCodec.h"
diff -r -u -N -w Main\include\auto\MogreDefaultHardwareBufferManager.h Main\include\auto\MogreDefaultHardwareBufferManager.h
--- Main\include\auto\MogreDefaultHardwareBufferManager.h	2013-04-22 20:21:38.248575200 +0200
+++ Main\include\auto\MogreDefaultHardwareBufferManager.h	2013-04-22 20:23:14.152161200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreDefaultHardwareBufferManager.h"
+#pragma managed(pop)
 #include "MogreHardwareBufferManager.h"
 #include "MogreHardwareBuffer.h"
 #include "MogreHardwareIndexBuffer.h"
diff -r -u -N -w Main\include\auto\MogreDistanceLodStrategy.h Main\include\auto\MogreDistanceLodStrategy.h
--- Main\include\auto\MogreDistanceLodStrategy.h	2013-04-22 20:21:38.267585600 +0200
+++ Main\include\auto\MogreDistanceLodStrategy.h	2013-04-22 20:23:14.157160200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreDistanceLodStrategy.h"
+#pragma managed(pop)
 #include "MogreLodStrategy.h"
 #include "MogrePrerequisites.h"
 #include "MogreMesh.h"
diff -r -u -N -w Main\include\auto\MogreEdgeListBuilder.h Main\include\auto\MogreEdgeListBuilder.h
--- Main\include\auto\MogreEdgeListBuilder.h	2013-04-22 20:21:38.280949200 +0200
+++ Main\include\auto\MogreEdgeListBuilder.h	2013-04-22 20:23:14.166165600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreEdgeListBuilder.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreEntity.h Main\include\auto\MogreEntity.h
--- Main\include\auto\MogreEntity.h	2013-04-22 20:21:38.349995100 +0200
+++ Main\include\auto\MogreEntity.h	2013-04-22 20:23:14.174244000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreEntity.h"
+#pragma managed(pop)
 #include "MogreMovableObject.h"
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
diff -r -u -N -w Main\include\auto\MogreException.h Main\include\auto\MogreException.h
--- Main\include\auto\MogreException.h	2013-04-22 20:21:38.364004100 +0200
+++ Main\include\auto\MogreException.h	2013-04-22 20:23:14.183250700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreException.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreExternalTextureSource.h Main\include\auto\MogreExternalTextureSource.h
--- Main\include\auto\MogreExternalTextureSource.h	2013-04-22 20:21:38.380015600 +0200
+++ Main\include\auto\MogreExternalTextureSource.h	2013-04-22 20:23:14.189255900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreExternalTextureSource.h"
+#pragma managed(pop)
 #include "MogreStringInterface.h"
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
diff -r -u -N -w Main\include\auto\MogreExternalTextureSourceManager.h Main\include\auto\MogreExternalTextureSourceManager.h
--- Main\include\auto\MogreExternalTextureSourceManager.h	2013-04-22 20:21:38.393024100 +0200
+++ Main\include\auto\MogreExternalTextureSourceManager.h	2013-04-22 20:23:14.197262500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreExternalTextureSourceManager.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreFileSystem.h Main\include\auto\MogreFileSystem.h
--- Main\include\auto\MogreFileSystem.h	2013-04-22 20:21:38.412035000 +0200
+++ Main\include\auto\MogreFileSystem.h	2013-04-22 20:23:14.202981200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreFileSystem.h"
+#pragma managed(pop)
 #include "MogreArchive.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreFont.h Main\include\auto\MogreFont.h
--- Main\include\auto\MogreFont.h	2013-04-22 20:21:38.437052700 +0200
+++ Main\include\auto\MogreFont.h	2013-04-22 20:23:14.208984400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreFont.h"
+#pragma managed(pop)
 #include "MogreResource.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreFontManager.h Main\include\auto\MogreFontManager.h
--- Main\include\auto\MogreFontManager.h	2013-04-22 20:21:38.465084200 +0200
+++ Main\include\auto\MogreFontManager.h	2013-04-22 20:23:14.215991300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreFontManager.h"
+#pragma managed(pop)
 #include "MogreResourceManager.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreFrameListener.h Main\include\auto\MogreFrameListener.h
--- Main\include\auto\MogreFrameListener.h	2013-04-22 20:21:38.470090600 +0200
+++ Main\include\auto\MogreFrameListener.h	2013-04-22 20:23:14.221995700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreFrameListener.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreFrustum.h Main\include\auto\MogreFrustum.h
--- Main\include\auto\MogreFrustum.h	2013-04-22 20:21:38.494101300 +0200
+++ Main\include\auto\MogreFrustum.h	2013-04-22 20:23:14.230999100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreFrustum.h"
+#pragma managed(pop)
 #include "MogreRenderable.h"
 #include "MogreMovableObject.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreGpuProgram.h Main\include\auto\MogreGpuProgram.h
--- Main\include\auto\MogreGpuProgram.h	2013-04-22 20:21:38.524689500 +0200
+++ Main\include\auto\MogreGpuProgram.h	2013-04-22 20:23:14.237005600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreGpuProgram.h"
+#pragma managed(pop)
 #include "MogreResource.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreGpuProgramManager.h Main\include\auto\MogreGpuProgramManager.h
--- Main\include\auto\MogreGpuProgramManager.h	2013-04-22 20:21:38.549710100 +0200
+++ Main\include\auto\MogreGpuProgramManager.h	2013-04-22 20:23:14.245010100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreGpuProgramManager.h"
+#pragma managed(pop)
 #include "MogreResourceManager.h"
 #include "MogrePrerequisites.h"
 #include "MogreGpuProgram.h"
diff -r -u -N -w Main\include\auto\MogreGpuProgramParams.h Main\include\auto\MogreGpuProgramParams.h
--- Main\include\auto\MogreGpuProgramParams.h	2013-04-22 20:21:38.645772300 +0200
+++ Main\include\auto\MogreGpuProgramParams.h	2013-04-22 20:23:14.257020200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreGpuProgramParams.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 #include "MogreSerializer.h"
diff -r -u -N -w Main\include\auto\MogreGrid2DPageStrategy.h Main\include\auto\MogreGrid2DPageStrategy.h
--- Main\include\auto\MogreGrid2DPageStrategy.h	2013-04-22 20:21:38.678796200 +0200
+++ Main\include\auto\MogreGrid2DPageStrategy.h	2013-04-22 20:23:14.268028200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreGrid2DPageStrategy.h"
+#pragma managed(pop)
 #include "MogrePageStrategy.h"
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreHardwareBuffer.h Main\include\auto\MogreHardwareBuffer.h
--- Main\include\auto\MogreHardwareBuffer.h	2013-04-22 20:21:38.689952400 +0200
+++ Main\include\auto\MogreHardwareBuffer.h	2013-04-22 20:23:14.277031100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreHardwareBuffer.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreHardwareBufferManager.h Main\include\auto\MogreHardwareBufferManager.h
--- Main\include\auto\MogreHardwareBufferManager.h	2013-04-22 20:21:38.699957800 +0200
+++ Main\include\auto\MogreHardwareBufferManager.h	2013-04-22 20:23:14.289047700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreHardwareBufferManager.h"
+#pragma managed(pop)
 #include "MogreHardwareBuffer.h"
 #include "MogreHardwareIndexBuffer.h"
 
diff -r -u -N -w Main\include\auto\MogreHardwareIndexBuffer.h Main\include\auto\MogreHardwareIndexBuffer.h
--- Main\include\auto\MogreHardwareIndexBuffer.h	2013-04-22 20:21:38.708968600 +0200
+++ Main\include\auto\MogreHardwareIndexBuffer.h	2013-04-22 20:23:14.301049200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreHardwareIndexBuffer.h"
+#pragma managed(pop)
 #include "MogreHardwareBuffer.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreHardwareOcclusionQuery.h Main\include\auto\MogreHardwareOcclusionQuery.h
--- Main\include\auto\MogreHardwareOcclusionQuery.h	2013-04-22 20:21:38.715970200 +0200
+++ Main\include\auto\MogreHardwareOcclusionQuery.h	2013-04-22 20:23:14.310053000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreHardwareOcclusionQuery.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreHardwarePixelBuffer.h Main\include\auto\MogreHardwarePixelBuffer.h
--- Main\include\auto\MogreHardwarePixelBuffer.h	2013-04-22 20:21:38.734849700 +0200
+++ Main\include\auto\MogreHardwarePixelBuffer.h	2013-04-22 20:23:14.318066900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreHardwarePixelBuffer.h"
+#pragma managed(pop)
 #include "MogreHardwareBuffer.h"
 #include "MogrePixelFormat.h"
 #include "MogreImage.h"
diff -r -u -N -w Main\include\auto\MogreHardwareVertexBuffer.h Main\include\auto\MogreHardwareVertexBuffer.h
--- Main\include\auto\MogreHardwareVertexBuffer.h	2013-04-22 20:21:38.762907000 +0200
+++ Main\include\auto\MogreHardwareVertexBuffer.h	2013-04-22 20:23:14.329065500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreHardwareVertexBuffer.h"
+#pragma managed(pop)
 #include "MogreHardwareBuffer.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreHighLevelGpuProgram.h Main\include\auto\MogreHighLevelGpuProgram.h
--- Main\include\auto\MogreHighLevelGpuProgram.h	2013-04-22 20:21:38.776914300 +0200
+++ Main\include\auto\MogreHighLevelGpuProgram.h	2013-04-22 20:23:14.341074700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreHighLevelGpuProgram.h"
+#pragma managed(pop)
 #include "MogreGpuProgram.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreHighLevelGpuProgramManager.h Main\include\auto\MogreHighLevelGpuProgramManager.h
--- Main\include\auto\MogreHighLevelGpuProgramManager.h	2013-04-22 20:21:38.786922600 +0200
+++ Main\include\auto\MogreHighLevelGpuProgramManager.h	2013-04-22 20:23:14.350083800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreHighLevelGpuProgramManager.h"
+#pragma managed(pop)
 #include "MogreResourceManager.h"
 #include "MogrePrerequisites.h"
 #include "MogreGpuProgram.h"
diff -r -u -N -w Main\include\auto\MogreImage.h Main\include\auto\MogreImage.h
--- Main\include\auto\MogreImage.h	2013-04-22 20:21:38.816943100 +0200
+++ Main\include\auto\MogreImage.h	2013-04-22 20:23:14.358085900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreImage.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePixelFormat.h"
 
diff -r -u -N -w Main\include\auto\MogreImageCodec.h Main\include\auto\MogreImageCodec.h
--- Main\include\auto\MogreImageCodec.h	2013-04-22 20:21:38.825948900 +0200
+++ Main\include\auto\MogreImageCodec.h	2013-04-22 20:23:14.366089600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreImageCodec.h"
+#pragma managed(pop)
 #include "MogreCodec.h"
 #include "MogrePrerequisites.h"
 #include "MogrePixelFormat.h"
diff -r -u -N -w Main\include\auto\MogreInstancedGeometry.h Main\include\auto\MogreInstancedGeometry.h
--- Main\include\auto\MogreInstancedGeometry.h	2013-04-22 20:21:38.879982600 +0200
+++ Main\include\auto\MogreInstancedGeometry.h	2013-04-22 20:23:14.376095500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreInstancedGeometry.h"
+#pragma managed(pop)
 #include "MogreMovableObject.h"
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
diff -r -u -N -w Main\include\auto\MogreKeyFrame.h Main\include\auto\MogreKeyFrame.h
--- Main\include\auto\MogreKeyFrame.h	2013-04-22 20:21:38.903240400 +0200
+++ Main\include\auto\MogreKeyFrame.h	2013-04-22 20:23:14.385098000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreKeyFrame.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreLight.h Main\include\auto\MogreLight.h
--- Main\include\auto\MogreLight.h	2013-04-22 20:21:38.949271100 +0200
+++ Main\include\auto\MogreLight.h	2013-04-22 20:23:14.393104200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreLight.h"
+#pragma managed(pop)
 #include "MogreMovableObject.h"
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
diff -r -u -N -w Main\include\auto\MogreLodListener.h Main\include\auto\MogreLodListener.h
--- Main\include\auto\MogreLodListener.h	2013-04-22 20:21:38.958278600 +0200
+++ Main\include\auto\MogreLodListener.h	2013-04-22 20:23:14.399109000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreLodListener.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreLodStrategy.h Main\include\auto\MogreLodStrategy.h
--- Main\include\auto\MogreLodStrategy.h	2013-04-22 20:21:38.967283200 +0200
+++ Main\include\auto\MogreLodStrategy.h	2013-04-22 20:23:14.407115200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreLodStrategy.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreMesh.h"
 #include "MogreMaterial.h"
diff -r -u -N -w Main\include\auto\MogreLodStrategyManager.h Main\include\auto\MogreLodStrategyManager.h
--- Main\include\auto\MogreLodStrategyManager.h	2013-04-22 20:21:38.977627300 +0200
+++ Main\include\auto\MogreLodStrategyManager.h	2013-04-22 20:23:14.416126300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreLodStrategyManager.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreLog.h Main\include\auto\MogreLog.h
--- Main\include\auto\MogreLog.h	2013-04-22 20:21:38.994639300 +0200
+++ Main\include\auto\MogreLog.h	2013-04-22 20:23:14.424125100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreLog.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreLogManager.h Main\include\auto\MogreLogManager.h
--- Main\include\auto\MogreLogManager.h	2013-04-22 20:21:39.006647700 +0200
+++ Main\include\auto\MogreLogManager.h	2013-04-22 20:23:14.431130000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreLogManager.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreLog.h"
 
diff -r -u -N -w Main\include\auto\MogreManualObject.h Main\include\auto\MogreManualObject.h
--- Main\include\auto\MogreManualObject.h	2013-04-22 20:21:39.071688900 +0200
+++ Main\include\auto\MogreManualObject.h	2013-04-22 20:23:14.441136300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreManualObject.h"
+#pragma managed(pop)
 #include "MogreMovableObject.h"
 #include "MogreRenderable.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreMaterial.h Main\include\auto\MogreMaterial.h
--- Main\include\auto\MogreMaterial.h	2013-04-22 20:21:39.119721400 +0200
+++ Main\include\auto\MogreMaterial.h	2013-04-22 20:23:14.453377500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreMaterial.h"
+#pragma managed(pop)
 #include "MogreResource.h"
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
diff -r -u -N -w Main\include\auto\MogreMaterialManager.h Main\include\auto\MogreMaterialManager.h
--- Main\include\auto\MogreMaterialManager.h	2013-04-22 20:21:39.141737800 +0200
+++ Main\include\auto\MogreMaterialManager.h	2013-04-22 20:23:14.462386600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreMaterialManager.h"
+#pragma managed(pop)
 #include "MogreResourceManager.h"
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
diff -r -u -N -w Main\include\auto\MogreMaterialSerializer.h Main\include\auto\MogreMaterialSerializer.h
--- Main\include\auto\MogreMaterialSerializer.h	2013-04-22 20:21:39.175758500 +0200
+++ Main\include\auto\MogreMaterialSerializer.h	2013-04-22 20:23:14.473392100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreMaterialSerializer.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreMesh.h Main\include\auto\MogreMesh.h
--- Main\include\auto\MogreMesh.h	2013-04-22 20:21:39.249810400 +0200
+++ Main\include\auto\MogreMesh.h	2013-04-22 20:23:14.483400800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreMesh.h"
+#pragma managed(pop)
 #include "MogreResource.h"
 #include "MogrePrerequisites.h"
 #include "MogreVertexBoneAssignment.h"
diff -r -u -N -w Main\include\auto\MogreMeshManager.h Main\include\auto\MogreMeshManager.h
--- Main\include\auto\MogreMeshManager.h	2013-04-22 20:21:39.296842500 +0200
+++ Main\include\auto\MogreMeshManager.h	2013-04-22 20:23:14.494406700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreMeshManager.h"
+#pragma managed(pop)
 #include "MogreResource.h"
 #include "MogreResourceManager.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreMeshSerializer.h Main\include\auto\MogreMeshSerializer.h
--- Main\include\auto\MogreMeshSerializer.h	2013-04-22 20:21:39.312850300 +0200
+++ Main\include\auto\MogreMeshSerializer.h	2013-04-22 20:23:14.501409600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreMeshSerializer.h"
+#pragma managed(pop)
 #include "MogreSerializer.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreMovableObject.h Main\include\auto\MogreMovableObject.h
--- Main\include\auto\MogreMovableObject.h	2013-04-22 20:21:39.328861000 +0200
+++ Main\include\auto\MogreMovableObject.h	2013-04-22 20:23:14.512417100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreMovableObject.h"
+#pragma managed(pop)
 #include "MogreAnimable.h"
 #include "MogreShadowCaster.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreMovablePlane.h Main\include\auto\MogreMovablePlane.h
--- Main\include\auto\MogreMovablePlane.h	2013-04-22 20:21:39.343122000 +0200
+++ Main\include\auto\MogreMovablePlane.h	2013-04-22 20:23:14.518419100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreMovablePlane.h"
+#pragma managed(pop)
 #include "MogreMovableObject.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreNode.h Main\include\auto\MogreNode.h
--- Main\include\auto\MogreNode.h	2013-04-22 20:21:39.360886900 +0200
+++ Main\include\auto\MogreNode.h	2013-04-22 20:23:14.529430300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreNode.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreNumerics.h Main\include\auto\MogreNumerics.h
--- Main\include\auto\MogreNumerics.h	2013-04-22 20:21:39.368890600 +0200
+++ Main\include\auto\MogreNumerics.h	2013-04-22 20:23:14.535431100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreNumerics.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreOverlay.h Main\include\auto\MogreOverlay.h
--- Main\include\auto\MogreOverlay.h	2013-04-22 20:21:39.384902600 +0200
+++ Main\include\auto\MogreOverlay.h	2013-04-22 20:23:14.542436000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreOverlay.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreOverlayContainer.h Main\include\auto\MogreOverlayContainer.h
--- Main\include\auto\MogreOverlayContainer.h	2013-04-22 20:21:39.401913700 +0200
+++ Main\include\auto\MogreOverlayContainer.h	2013-04-22 20:23:14.549442500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreOverlayContainer.h"
+#pragma managed(pop)
 #include "MogreOverlayElement.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreOverlayElement.h Main\include\auto\MogreOverlayElement.h
--- Main\include\auto\MogreOverlayElement.h	2013-04-22 20:21:39.424927300 +0200
+++ Main\include\auto\MogreOverlayElement.h	2013-04-22 20:23:14.559014000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreOverlayElement.h"
+#pragma managed(pop)
 #include "MogreStringInterface.h"
 #include "MogreRenderable.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreOverlayElementFactory.h Main\include\auto\MogreOverlayElementFactory.h
--- Main\include\auto\MogreOverlayElementFactory.h	2013-04-22 20:21:39.441938800 +0200
+++ Main\include\auto\MogreOverlayElementFactory.h	2013-04-22 20:23:14.566017700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreOverlayElementFactory.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreOverlayManager.h Main\include\auto\MogreOverlayManager.h
--- Main\include\auto\MogreOverlayManager.h	2013-04-22 20:21:39.480966800 +0200
+++ Main\include\auto\MogreOverlayManager.h	2013-04-22 20:23:14.574023400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreOverlayManager.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreFrustum.h"
 #include "MogreStringVector.h"
diff -r -u -N -w Main\include\auto\MogrePage.h Main\include\auto\MogrePage.h
--- Main\include\auto\MogrePage.h	2013-04-22 20:21:39.503979900 +0200
+++ Main\include\auto\MogrePage.h	2013-04-22 20:23:14.582027500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePage.h"
+#pragma managed(pop)
 #include "MogreWorkQueue.h"
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogrePageContent.h Main\include\auto\MogrePageContent.h
--- Main\include\auto\MogrePageContent.h	2013-04-22 20:21:39.515988700 +0200
+++ Main\include\auto\MogrePageContent.h	2013-04-22 20:23:14.590263600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePageContent.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogrePageContentCollection.h Main\include\auto\MogrePageContentCollection.h
--- Main\include\auto\MogrePageContentCollection.h	2013-04-22 20:21:39.524994100 +0200
+++ Main\include\auto\MogrePageContentCollection.h	2013-04-22 20:23:14.598269000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePageContentCollection.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogrePageContentCollectionFactory.h Main\include\auto\MogrePageContentCollectionFactory.h
--- Main\include\auto\MogrePageContentCollectionFactory.h	2013-04-22 20:21:39.531001400 +0200
+++ Main\include\auto\MogrePageContentCollectionFactory.h	2013-04-22 20:23:14.604271700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePageContentCollectionFactory.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogrePageContentFactory.h Main\include\auto\MogrePageContentFactory.h
--- Main\include\auto\MogrePageContentFactory.h	2013-04-22 20:21:39.536012800 +0200
+++ Main\include\auto\MogrePageContentFactory.h	2013-04-22 20:23:14.614279700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePageContentFactory.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogrePagedWorld.h Main\include\auto\MogrePagedWorld.h
--- Main\include\auto\MogrePagedWorld.h	2013-04-22 20:21:39.558029100 +0200
+++ Main\include\auto\MogrePagedWorld.h	2013-04-22 20:23:14.625285600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePagedWorld.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
 
diff -r -u -N -w Main\include\auto\MogrePagedWorldSection.h Main\include\auto\MogrePagedWorldSection.h
--- Main\include\auto\MogrePagedWorldSection.h	2013-04-22 20:21:39.593042100 +0200
+++ Main\include\auto\MogrePagedWorldSection.h	2013-04-22 20:23:14.633290500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePagedWorldSection.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogrePageManager.h Main\include\auto\MogrePageManager.h
--- Main\include\auto\MogrePageManager.h	2013-04-22 20:21:39.635072200 +0200
+++ Main\include\auto\MogrePageManager.h	2013-04-22 20:23:14.642299200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePageManager.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
 
diff -r -u -N -w Main\include\auto\MogrePageStrategy.h Main\include\auto\MogrePageStrategy.h
--- Main\include\auto\MogrePageStrategy.h	2013-04-22 20:21:39.646080200 +0200
+++ Main\include\auto\MogrePageStrategy.h	2013-04-22 20:23:14.651303000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePageStrategy.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
 
diff -r -u -N -w Main\include\auto\MogrePagingPrerequisites.h Main\include\auto\MogrePagingPrerequisites.h
--- Main\include\auto\MogrePagingPrerequisites.h	2013-04-22 20:21:39.650084500 +0200
+++ Main\include\auto\MogrePagingPrerequisites.h	2013-04-22 20:23:14.657559500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePagingPrerequisites.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogrePanelOverlayElement.h Main\include\auto\MogrePanelOverlayElement.h
--- Main\include\auto\MogrePanelOverlayElement.h	2013-04-22 20:21:39.657085700 +0200
+++ Main\include\auto\MogrePanelOverlayElement.h	2013-04-22 20:23:14.664566800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePanelOverlayElement.h"
+#pragma managed(pop)
 #include "MogreOverlayContainer.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreParticle.h Main\include\auto\MogreParticle.h
--- Main\include\auto\MogreParticle.h	2013-04-22 20:21:39.666093600 +0200
+++ Main\include\auto\MogreParticle.h	2013-04-22 20:23:14.673568900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreParticle.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreParticleAffector.h Main\include\auto\MogreParticleAffector.h
--- Main\include\auto\MogreParticleAffector.h	2013-04-22 20:21:39.673098500 +0200
+++ Main\include\auto\MogreParticleAffector.h	2013-04-22 20:23:14.681574300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreParticleAffector.h"
+#pragma managed(pop)
 #include "MogreStringInterface.h"
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
diff -r -u -N -w Main\include\auto\MogreParticleAffectorFactory.h Main\include\auto\MogreParticleAffectorFactory.h
--- Main\include\auto\MogreParticleAffectorFactory.h	2013-04-22 20:21:39.681106300 +0200
+++ Main\include\auto\MogreParticleAffectorFactory.h	2013-04-22 20:23:14.689585000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreParticleAffectorFactory.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreParticleEmitter.h Main\include\auto\MogreParticleEmitter.h
--- Main\include\auto\MogreParticleEmitter.h	2013-04-22 20:21:39.712123900 +0200
+++ Main\include\auto\MogreParticleEmitter.h	2013-04-22 20:23:14.698586300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreParticleEmitter.h"
+#pragma managed(pop)
 #include "MogreStringInterface.h"
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
diff -r -u -N -w Main\include\auto\MogreParticleEmitterFactory.h Main\include\auto\MogreParticleEmitterFactory.h
--- Main\include\auto\MogreParticleEmitterFactory.h	2013-04-22 20:21:39.720131700 +0200
+++ Main\include\auto\MogreParticleEmitterFactory.h	2013-04-22 20:23:14.708591700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreParticleEmitterFactory.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreParticleIterator.h Main\include\auto\MogreParticleIterator.h
--- Main\include\auto\MogreParticleIterator.h	2013-04-22 20:21:39.725132800 +0200
+++ Main\include\auto\MogreParticleIterator.h	2013-04-22 20:23:14.715596600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreParticleIterator.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreParticleSystem.h Main\include\auto\MogreParticleSystem.h
--- Main\include\auto\MogreParticleSystem.h	2013-04-22 20:21:39.797844400 +0200
+++ Main\include\auto\MogreParticleSystem.h	2013-04-22 20:23:14.723449600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreParticleSystem.h"
+#pragma managed(pop)
 #include "MogreStringInterface.h"
 #include "MogreMovableObject.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreParticleSystemManager.h Main\include\auto\MogreParticleSystemManager.h
--- Main\include\auto\MogreParticleSystemManager.h	2013-04-22 20:21:39.836867000 +0200
+++ Main\include\auto\MogreParticleSystemManager.h	2013-04-22 20:23:14.731455400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreParticleSystemManager.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreStringVector.h"
 #include "MogreMovableObject.h"
diff -r -u -N -w Main\include\auto\MogreParticleSystemRenderer.h Main\include\auto\MogreParticleSystemRenderer.h
--- Main\include\auto\MogreParticleSystemRenderer.h	2013-04-22 20:21:39.849879200 +0200
+++ Main\include\auto\MogreParticleSystemRenderer.h	2013-04-22 20:23:14.739459500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreParticleSystemRenderer.h"
+#pragma managed(pop)
 #include "MogreStringInterface.h"
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
diff -r -u -N -w Main\include\auto\MogrePass.h Main\include\auto\MogrePass.h
--- Main\include\auto\MogrePass.h	2013-04-22 20:21:39.931929500 +0200
+++ Main\include\auto\MogrePass.h	2013-04-22 20:23:14.750466700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePass.h"
+#pragma managed(pop)
 #include "MogreCommon.h"
 #include "MogreBlendMode.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogrePatchMesh.h Main\include\auto\MogrePatchMesh.h
--- Main\include\auto\MogrePatchMesh.h	2013-04-22 20:21:39.947939800 +0200
+++ Main\include\auto\MogrePatchMesh.h	2013-04-22 20:23:14.757471100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePatchMesh.h"
+#pragma managed(pop)
 #include "MogreMesh.h"
 #include "MogrePrerequisites.h"
 #include "MogreResource.h"
diff -r -u -N -w Main\include\auto\MogrePatchSurface.h Main\include\auto\MogrePatchSurface.h
--- Main\include\auto\MogrePatchSurface.h	2013-04-22 20:21:39.962949200 +0200
+++ Main\include\auto\MogrePatchSurface.h	2013-04-22 20:23:14.767479500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePatchSurface.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogrePixelCountLodStrategy.h Main\include\auto\MogrePixelCountLodStrategy.h
--- Main\include\auto\MogrePixelCountLodStrategy.h	2013-04-22 20:21:39.972026000 +0200
+++ Main\include\auto\MogrePixelCountLodStrategy.h	2013-04-22 20:23:14.773482300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePixelCountLodStrategy.h"
+#pragma managed(pop)
 #include "MogreLodStrategy.h"
 #include "MogrePrerequisites.h"
 #include "MogreMesh.h"
diff -r -u -N -w Main\include\auto\MogrePixelFormat.h Main\include\auto\MogrePixelFormat.h
--- Main\include\auto\MogrePixelFormat.h	2013-04-22 20:21:39.993040700 +0200
+++ Main\include\auto\MogrePixelFormat.h	2013-04-22 20:23:14.782488100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePixelFormat.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
 
diff -r -u -N -w Main\include\auto\MogrePlane.h Main\include\auto\MogrePlane.h
--- Main\include\auto\MogrePlane.h	2013-04-22 20:21:39.997043400 +0200
+++ Main\include\auto\MogrePlane.h	2013-04-22 20:23:14.791495900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePlane.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogrePlaneBoundedVolume.h Main\include\auto\MogrePlaneBoundedVolume.h
--- Main\include\auto\MogrePlaneBoundedVolume.h	2013-04-22 20:21:40.006052000 +0200
+++ Main\include\auto\MogrePlaneBoundedVolume.h	2013-04-22 20:23:14.798501600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePlaneBoundedVolume.h"
+#pragma managed(pop)
 #include "MogrePlane.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogrePlatform.h Main\include\auto\MogrePlatform.h
--- Main\include\auto\MogrePlatform.h	2013-04-22 20:21:40.013054100 +0200
+++ Main\include\auto\MogrePlatform.h	2013-04-22 20:23:14.807506200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePlatform.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogrePlugin.h Main\include\auto\MogrePlugin.h
--- Main\include\auto\MogrePlugin.h	2013-04-22 20:21:40.024062000 +0200
+++ Main\include\auto\MogrePlugin.h	2013-04-22 20:23:14.815509500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePlugin.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogrePose.h Main\include\auto\MogrePose.h
--- Main\include\auto\MogrePose.h	2013-04-22 20:21:40.034067900 +0200
+++ Main\include\auto\MogrePose.h	2013-04-22 20:23:14.823516900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePose.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogrePrerequisites.h Main\include\auto\MogrePrerequisites.h
--- Main\include\auto\MogrePrerequisites.h	2013-04-22 20:21:40.044076200 +0200
+++ Main\include\auto\MogrePrerequisites.h	2013-04-22 20:23:14.831524700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePrerequisites.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreProfiler.h Main\include\auto\MogreProfiler.h
--- Main\include\auto\MogreProfiler.h	2013-04-22 20:21:40.064089200 +0200
+++ Main\include\auto\MogreProfiler.h	2013-04-22 20:23:14.839526800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreProfiler.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
 
diff -r -u -N -w Main\include\auto\MogreProgressiveMesh.h Main\include\auto\MogreProgressiveMesh.h
--- Main\include\auto\MogreProgressiveMesh.h	2013-04-22 20:21:40.074095100 +0200
+++ Main\include\auto\MogreProgressiveMesh.h	2013-04-22 20:23:14.848532200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreProgressiveMesh.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogrePVRTCCodec.h Main\include\auto\MogrePVRTCCodec.h
--- Main\include\auto\MogrePVRTCCodec.h	2013-04-22 20:21:40.086103100 +0200
+++ Main\include\auto\MogrePVRTCCodec.h	2013-04-22 20:23:14.864546200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgrePVRTCCodec.h"
+#pragma managed(pop)
 #include "MogreImageCodec.h"
 #include "MogrePrerequisites.h"
 #include "MogreCodec.h"
diff -r -u -N -w Main\include\auto\MogreRectangle2D.h Main\include\auto\MogreRectangle2D.h
--- Main\include\auto\MogreRectangle2D.h	2013-04-22 20:21:40.105113100 +0200
+++ Main\include\auto\MogreRectangle2D.h	2013-04-22 20:23:14.872547800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRectangle2D.h"
+#pragma managed(pop)
 #include "MogreSimpleRenderable.h"
 #include "MogreHardwareBuffer.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreRenderable.h Main\include\auto\MogreRenderable.h
--- Main\include\auto\MogreRenderable.h	2013-04-22 20:21:40.115289400 +0200
+++ Main\include\auto\MogreRenderable.h	2013-04-22 20:23:14.888559800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderable.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreGpuProgramParams.h"
 
diff -r -u -N -w Main\include\auto\MogreRenderObjectListener.h Main\include\auto\MogreRenderObjectListener.h
--- Main\include\auto\MogreRenderObjectListener.h	2013-04-22 20:21:40.122294200 +0200
+++ Main\include\auto\MogreRenderObjectListener.h	2013-04-22 20:23:14.904571300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderObjectListener.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreRenderOperation.h Main\include\auto\MogreRenderOperation.h
--- Main\include\auto\MogreRenderOperation.h	2013-04-22 20:21:40.128299900 +0200
+++ Main\include\auto\MogreRenderOperation.h	2013-04-22 20:23:14.928636200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderOperation.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreRenderQueue.h Main\include\auto\MogreRenderQueue.h
--- Main\include\auto\MogreRenderQueue.h	2013-04-22 20:21:40.144309400 +0200
+++ Main\include\auto\MogreRenderQueue.h	2013-04-22 20:23:14.937635400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderQueue.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 #include "MogreSceneManager.h"
diff -r -u -N -w Main\include\auto\MogreRenderQueueInvocation.h Main\include\auto\MogreRenderQueueInvocation.h
--- Main\include\auto\MogreRenderQueueInvocation.h	2013-04-22 20:21:40.161320500 +0200
+++ Main\include\auto\MogreRenderQueueInvocation.h	2013-04-22 20:23:14.945642900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderQueueInvocation.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 #include "MogreRenderQueueSortingGrouping.h"
diff -r -u -N -w Main\include\auto\MogreRenderQueueListener.h Main\include\auto\MogreRenderQueueListener.h
--- Main\include\auto\MogreRenderQueueListener.h	2013-04-22 20:21:40.170328000 +0200
+++ Main\include\auto\MogreRenderQueueListener.h	2013-04-22 20:23:14.954647400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderQueueListener.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreRenderQueueSortingGrouping.h Main\include\auto\MogreRenderQueueSortingGrouping.h
--- Main\include\auto\MogreRenderQueueSortingGrouping.h	2013-04-22 20:21:40.197343000 +0200
+++ Main\include\auto\MogreRenderQueueSortingGrouping.h	2013-04-22 20:23:14.966162000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderQueueSortingGrouping.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreRenderSystem.h Main\include\auto\MogreRenderSystem.h
--- Main\include\auto\MogreRenderSystem.h	2013-04-22 20:21:40.272395400 +0200
+++ Main\include\auto\MogreRenderSystem.h	2013-04-22 20:23:14.979172500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderSystem.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
 #include "MogreHardwareVertexBuffer.h"
diff -r -u -N -w Main\include\auto\MogreRenderSystemCapabilities.h Main\include\auto\MogreRenderSystemCapabilities.h
--- Main\include\auto\MogreRenderSystemCapabilities.h	2013-04-22 20:21:40.303416700 +0200
+++ Main\include\auto\MogreRenderSystemCapabilities.h	2013-04-22 20:23:14.988177100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderSystemCapabilities.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreRenderSystemCapabilitiesManager.h Main\include\auto\MogreRenderSystemCapabilitiesManager.h
--- Main\include\auto\MogreRenderSystemCapabilitiesManager.h	2013-04-22 20:21:40.328945600 +0200
+++ Main\include\auto\MogreRenderSystemCapabilitiesManager.h	2013-04-22 20:23:14.997182100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderSystemCapabilitiesManager.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreRenderTarget.h Main\include\auto\MogreRenderTarget.h
--- Main\include\auto\MogreRenderTarget.h	2013-04-22 20:21:40.358966500 +0200
+++ Main\include\auto\MogreRenderTarget.h	2013-04-22 20:23:15.006185500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderTarget.h"
+#pragma managed(pop)
 #include "MogreRenderTargetListener.h"
 #include "MogrePrerequisites.h"
 #include "MogrePixelFormat.h"
diff -r -u -N -w Main\include\auto\MogreRenderTargetListener.h Main\include\auto\MogreRenderTargetListener.h
--- Main\include\auto\MogreRenderTargetListener.h	2013-04-22 20:21:40.364238900 +0200
+++ Main\include\auto\MogreRenderTargetListener.h	2013-04-22 20:23:15.014190400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderTargetListener.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreRenderTexture.h Main\include\auto\MogreRenderTexture.h
--- Main\include\auto\MogreRenderTexture.h	2013-04-22 20:21:40.377247800 +0200
+++ Main\include\auto\MogreRenderTexture.h	2013-04-22 20:23:15.024198700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderTexture.h"
+#pragma managed(pop)
 #include "MogreRenderTarget.h"
 #include "MogrePixelFormat.h"
 
diff -r -u -N -w Main\include\auto\MogreRenderToVertexBuffer.h Main\include\auto\MogreRenderToVertexBuffer.h
--- Main\include\auto\MogreRenderToVertexBuffer.h	2013-04-22 20:21:40.388255800 +0200
+++ Main\include\auto\MogreRenderToVertexBuffer.h	2013-04-22 20:23:15.032202000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderToVertexBuffer.h"
+#pragma managed(pop)
 #include "MogreRenderOperation.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreRenderWindow.h Main\include\auto\MogreRenderWindow.h
--- Main\include\auto\MogreRenderWindow.h	2013-04-22 20:21:40.407269900 +0200
+++ Main\include\auto\MogreRenderWindow.h	2013-04-22 20:23:15.041208300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRenderWindow.h"
+#pragma managed(pop)
 #include "MogreRenderTarget.h"
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
diff -r -u -N -w Main\include\auto\MogreResource.h Main\include\auto\MogreResource.h
--- Main\include\auto\MogreResource.h	2013-04-22 20:21:40.419275500 +0200
+++ Main\include\auto\MogreResource.h	2013-04-22 20:23:15.053217900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreResource.h"
+#pragma managed(pop)
 #include "MogreStringInterface.h"
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
diff -r -u -N -w Main\include\auto\MogreResourceBackgroundQueue.h Main\include\auto\MogreResourceBackgroundQueue.h
--- Main\include\auto\MogreResourceBackgroundQueue.h	2013-04-22 20:21:40.428286200 +0200
+++ Main\include\auto\MogreResourceBackgroundQueue.h	2013-04-22 20:23:15.060223600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreResourceBackgroundQueue.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreResourceGroupManager.h Main\include\auto\MogreResourceGroupManager.h
--- Main\include\auto\MogreResourceGroupManager.h	2013-04-22 20:21:40.508334400 +0200
+++ Main\include\auto\MogreResourceGroupManager.h	2013-04-22 20:23:15.071231600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreResourceGroupManager.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
 #include "MogreStringVector.h"
diff -r -u -N -w Main\include\auto\MogreResourceManager.h Main\include\auto\MogreResourceManager.h
--- Main\include\auto\MogreResourceManager.h	2013-04-22 20:21:40.534352600 +0200
+++ Main\include\auto\MogreResourceManager.h	2013-04-22 20:23:15.079057100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreResourceManager.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreResource.h"
 #include "MogreCommon.h"
diff -r -u -N -w Main\include\auto\MogreRibbonTrail.h Main\include\auto\MogreRibbonTrail.h
--- Main\include\auto\MogreRibbonTrail.h	2013-04-22 20:21:40.582384300 +0200
+++ Main\include\auto\MogreRibbonTrail.h	2013-04-22 20:23:15.086064100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRibbonTrail.h"
+#pragma managed(pop)
 #include "MogreNode.h"
 #include "MogreBillboardChain.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreRoot.h Main\include\auto\MogreRoot.h
--- Main\include\auto\MogreRoot.h	2013-04-22 20:21:40.643423600 +0200
+++ Main\include\auto\MogreRoot.h	2013-04-22 20:29:22.258816200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRoot.h"
+#pragma managed(pop)
 #include "MogreFrameListener.h"
 #include "MogrePrerequisites.h"
 #include "MogreSceneManager.h"
@@ -58,12 +60,9 @@
     {
     }
   
+    // Destructor - this runs in the main thread -- if the user calls Dispose()
     ~Root()
     {
-      this->!Root();
-    }
-    !Root()
-    {
       _native = Ogre::Root::getSingletonPtr();
       
       // Collect all SharedPtr objects that are waiting for finalization
@@ -80,6 +79,10 @@
       _singleton = nullptr;
     }
   
+    // Finalizer - this runs in the garbage collector thread.
+    !Root()
+    {
+    }
   
     //Public Declarations
   public:
diff -r -u -N -w Main\include\auto\MogreRotationalSpline.h Main\include\auto\MogreRotationalSpline.h
--- Main\include\auto\MogreRotationalSpline.h	2013-04-22 20:21:40.655433700 +0200
+++ Main\include\auto\MogreRotationalSpline.h	2013-04-22 20:23:15.115086100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreRotationalSpline.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreSceneManager.h Main\include\auto\MogreSceneManager.h
--- Main\include\auto\MogreSceneManager.h	2013-04-22 20:21:40.872577000 +0200
+++ Main\include\auto\MogreSceneManager.h	2013-04-22 20:23:15.125088700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSceneManager.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreRenderQueueListener.h"
 #include "MogrePlatform.h"
diff -r -u -N -w Main\include\auto\MogreSceneManagerEnumerator.h Main\include\auto\MogreSceneManagerEnumerator.h
--- Main\include\auto\MogreSceneManagerEnumerator.h	2013-04-22 20:21:40.902596700 +0200
+++ Main\include\auto\MogreSceneManagerEnumerator.h	2013-04-22 20:23:15.132092800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSceneManagerEnumerator.h"
+#pragma managed(pop)
 #include "MogreSceneManager.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreSceneNode.h Main\include\auto\MogreSceneNode.h
--- Main\include\auto\MogreSceneNode.h	2013-04-22 20:21:40.941625400 +0200
+++ Main\include\auto\MogreSceneNode.h	2013-04-22 20:23:15.143100800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSceneNode.h"
+#pragma managed(pop)
 #include "MogreNode.h"
 #include "MogrePrerequisites.h"
 #include "MogreSceneManager.h"
diff -r -u -N -w Main\include\auto\MogreSceneQuery.h Main\include\auto\MogreSceneQuery.h
--- Main\include\auto\MogreSceneQuery.h	2013-04-22 20:21:40.964640600 +0200
+++ Main\include\auto\MogreSceneQuery.h	2013-04-22 20:23:15.150691900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSceneQuery.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePlaneBoundedVolume.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreSerializer.h Main\include\auto\MogreSerializer.h
--- Main\include\auto\MogreSerializer.h	2013-04-22 20:21:40.973648900 +0200
+++ Main\include\auto\MogreSerializer.h	2013-04-22 20:23:15.158116700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSerializer.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreShadowCameraSetup.h Main\include\auto\MogreShadowCameraSetup.h
--- Main\include\auto\MogreShadowCameraSetup.h	2013-04-22 20:21:40.982652600 +0200
+++ Main\include\auto\MogreShadowCameraSetup.h	2013-04-22 20:23:15.168123400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreShadowCameraSetup.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreShadowCameraSetupFocused.h Main\include\auto\MogreShadowCameraSetupFocused.h
--- Main\include\auto\MogreShadowCameraSetupFocused.h	2013-04-22 20:21:40.994660200 +0200
+++ Main\include\auto\MogreShadowCameraSetupFocused.h	2013-04-22 20:23:15.176129600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreShadowCameraSetupFocused.h"
+#pragma managed(pop)
 #include "MogreShadowCameraSetup.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreShadowCameraSetupLiSPSM.h Main\include\auto\MogreShadowCameraSetupLiSPSM.h
--- Main\include\auto\MogreShadowCameraSetupLiSPSM.h	2013-04-22 20:21:41.009668400 +0200
+++ Main\include\auto\MogreShadowCameraSetupLiSPSM.h	2013-04-22 20:23:15.183013800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreShadowCameraSetupLiSPSM.h"
+#pragma managed(pop)
 #include "MogreShadowCameraSetupFocused.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreShadowCameraSetupPlaneOptimal.h Main\include\auto\MogreShadowCameraSetupPlaneOptimal.h
--- Main\include\auto\MogreShadowCameraSetupPlaneOptimal.h	2013-04-22 20:21:41.015678600 +0200
+++ Main\include\auto\MogreShadowCameraSetupPlaneOptimal.h	2013-04-22 20:23:15.189019000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreShadowCameraSetupPlaneOptimal.h"
+#pragma managed(pop)
 #include "MogreShadowCameraSetup.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreShadowCameraSetupPSSM.h Main\include\auto\MogreShadowCameraSetupPSSM.h
--- Main\include\auto\MogreShadowCameraSetupPSSM.h	2013-04-22 20:21:41.030684700 +0200
+++ Main\include\auto\MogreShadowCameraSetupPSSM.h	2013-04-22 20:23:15.197033000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreShadowCameraSetupPSSM.h"
+#pragma managed(pop)
 #include "MogreShadowCameraSetupLiSPSM.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreShadowCaster.h Main\include\auto\MogreShadowCaster.h
--- Main\include\auto\MogreShadowCaster.h	2013-04-22 20:21:41.040695100 +0200
+++ Main\include\auto\MogreShadowCaster.h	2013-04-22 20:23:15.204028500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreShadowCaster.h"
+#pragma managed(pop)
 #include "MogreRenderable.h"
 #include "MogrePrerequisites.h"
 #include "MogreGpuProgramParams.h"
diff -r -u -N -w Main\include\auto\MogreShadowTextureManager.h Main\include\auto\MogreShadowTextureManager.h
--- Main\include\auto\MogreShadowTextureManager.h	2013-04-22 20:21:41.050698100 +0200
+++ Main\include\auto\MogreShadowTextureManager.h	2013-04-22 20:23:15.215034800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreShadowTextureManager.h"
+#pragma managed(pop)
 #include "MogrePixelFormat.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreSimplePageContentCollection.h Main\include\auto\MogreSimplePageContentCollection.h
--- Main\include\auto\MogreSimplePageContentCollection.h	2013-04-22 20:21:41.063704500 +0200
+++ Main\include\auto\MogreSimplePageContentCollection.h	2013-04-22 20:23:15.223041000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSimplePageContentCollection.h"
+#pragma managed(pop)
 #include "MogrePageContentCollection.h"
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreSimpleRenderable.h Main\include\auto\MogreSimpleRenderable.h
--- Main\include\auto\MogreSimpleRenderable.h	2013-04-22 20:21:41.071712700 +0200
+++ Main\include\auto\MogreSimpleRenderable.h	2013-04-22 20:23:15.230043400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSimpleRenderable.h"
+#pragma managed(pop)
 #include "MogreRenderable.h"
 #include "MogreMovableObject.h"
 #include "MogrePrerequisites.h"
diff -r -u -N -w Main\include\auto\MogreSimpleSpline.h Main\include\auto\MogreSimpleSpline.h
--- Main\include\auto\MogreSimpleSpline.h	2013-04-22 20:21:41.080719400 +0200
+++ Main\include\auto\MogreSimpleSpline.h	2013-04-22 20:23:15.237054400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSimpleSpline.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreSkeleton.h Main\include\auto\MogreSkeleton.h
--- Main\include\auto\MogreSkeleton.h	2013-04-22 20:21:41.114740900 +0200
+++ Main\include\auto\MogreSkeleton.h	2013-04-22 20:23:15.247057400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSkeleton.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreResource.h"
 #include "MogreStringVector.h"
diff -r -u -N -w Main\include\auto\MogreSkeletonInstance.h Main\include\auto\MogreSkeletonInstance.h
--- Main\include\auto\MogreSkeletonInstance.h	2013-04-22 20:21:41.134753900 +0200
+++ Main\include\auto\MogreSkeletonInstance.h	2013-04-22 20:23:15.254059400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSkeletonInstance.h"
+#pragma managed(pop)
 #include "MogreSkeleton.h"
 #include "MogrePrerequisites.h"
 #include "MogreResource.h"
diff -r -u -N -w Main\include\auto\MogreSkeletonManager.h Main\include\auto\MogreSkeletonManager.h
--- Main\include\auto\MogreSkeletonManager.h	2013-04-22 20:21:41.143166500 +0200
+++ Main\include\auto\MogreSkeletonManager.h	2013-04-22 20:23:15.261064300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSkeletonManager.h"
+#pragma managed(pop)
 #include "MogreResourceManager.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreSkeletonSerializer.h Main\include\auto\MogreSkeletonSerializer.h
--- Main\include\auto\MogreSkeletonSerializer.h	2013-04-22 20:21:41.162179400 +0200
+++ Main\include\auto\MogreSkeletonSerializer.h	2013-04-22 20:23:15.270072200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSkeletonSerializer.h"
+#pragma managed(pop)
 #include "MogreSerializer.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreStaticGeometry.h Main\include\auto\MogreStaticGeometry.h
--- Main\include\auto\MogreStaticGeometry.h	2013-04-22 20:21:41.212210700 +0200
+++ Main\include\auto\MogreStaticGeometry.h	2013-04-22 20:23:15.280077600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreStaticGeometry.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
 #include "MogreShadowCaster.h"
diff -r -u -N -w Main\include\auto\MogreStreamSerialiser.h Main\include\auto\MogreStreamSerialiser.h
--- Main\include\auto\MogreStreamSerialiser.h	2013-04-22 20:21:41.235227200 +0200
+++ Main\include\auto\MogreStreamSerialiser.h	2013-04-22 20:23:15.286080400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreStreamSerialiser.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreString.h Main\include\auto\MogreString.h
--- Main\include\auto\MogreString.h	2013-04-22 20:21:41.240218800 +0200
+++ Main\include\auto\MogreString.h	2013-04-22 20:23:15.292088500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreString.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreStringConverter.h Main\include\auto\MogreStringConverter.h
--- Main\include\auto\MogreStringConverter.h	2013-04-22 20:21:41.267246100 +0200
+++ Main\include\auto\MogreStringConverter.h	2013-04-22 20:23:15.299095900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreStringConverter.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreStringVector.h"
 
diff -r -u -N -w Main\include\auto\MogreStringInterface.h Main\include\auto\MogreStringInterface.h
--- Main\include\auto\MogreStringInterface.h	2013-04-22 20:21:41.277252800 +0200
+++ Main\include\auto\MogreStringInterface.h	2013-04-22 20:23:15.308098400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreStringInterface.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
 
diff -r -u -N -w Main\include\auto\MogreStringVector.h Main\include\auto\MogreStringVector.h
--- Main\include\auto\MogreStringVector.h	2013-04-22 20:21:41.284260100 +0200
+++ Main\include\auto\MogreStringVector.h	2013-04-22 20:23:15.317106300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreStringVector.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreSubEntity.h Main\include\auto\MogreSubEntity.h
--- Main\include\auto\MogreSubEntity.h	2013-04-22 20:21:41.299269100 +0200
+++ Main\include\auto\MogreSubEntity.h	2013-04-22 20:23:15.326107600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSubEntity.h"
+#pragma managed(pop)
 #include "MogreRenderable.h"
 #include "MogrePrerequisites.h"
 #include "MogreGpuProgramParams.h"
diff -r -u -N -w Main\include\auto\MogreSubMesh.h Main\include\auto\MogreSubMesh.h
--- Main\include\auto\MogreSubMesh.h	2013-04-22 20:21:41.320282200 +0200
+++ Main\include\auto\MogreSubMesh.h	2013-04-22 20:23:15.335113800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreSubMesh.h"
+#pragma managed(pop)
 #include "MogreVertexBoneAssignment.h"
 #include "MogrePrerequisites.h"
 #include "MogreRenderOperation.h"
diff -r -u -N -w Main\include\auto\MogreTagPoint.h Main\include\auto\MogreTagPoint.h
--- Main\include\auto\MogreTagPoint.h	2013-04-22 20:21:41.337294500 +0200
+++ Main\include\auto\MogreTagPoint.h	2013-04-22 20:23:15.344120000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTagPoint.h"
+#pragma managed(pop)
 #include "MogreBone.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreTechnique.h Main\include\auto\MogreTechnique.h
--- Main\include\auto\MogreTechnique.h	2013-04-22 20:21:41.376320800 +0200
+++ Main\include\auto\MogreTechnique.h	2013-04-22 20:23:15.355129200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTechnique.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreRenderSystemCapabilities.h"
 #include "MogrePass.h"
diff -r -u -N -w Main\include\auto\MogreTerrain.h Main\include\auto\MogreTerrain.h
--- Main\include\auto\MogreTerrain.h	2013-04-22 20:21:41.526387100 +0200
+++ Main\include\auto\MogreTerrain.h	2013-04-22 20:23:15.366135200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTerrain.h"
+#pragma managed(pop)
 #include "MogreSceneManager.h"
 #include "MogreWorkQueue.h"
 #include "MogrePlatform.h"
diff -r -u -N -w Main\include\auto\MogreTerrainGroup.h Main\include\auto\MogreTerrainGroup.h
--- Main\include\auto\MogreTerrainGroup.h	2013-04-22 20:21:41.574420800 +0200
+++ Main\include\auto\MogreTerrainGroup.h	2013-04-22 20:23:15.375146300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTerrainGroup.h"
+#pragma managed(pop)
 #include "MogreWorkQueue.h"
 #include "MogrePrerequisites.h"
 #include "MogreTerrain.h"
diff -r -u -N -w Main\include\auto\MogreTerrainLayerBlendMap.h Main\include\auto\MogreTerrainLayerBlendMap.h
--- Main\include\auto\MogreTerrainLayerBlendMap.h	2013-04-22 20:21:41.593433300 +0200
+++ Main\include\auto\MogreTerrainLayerBlendMap.h	2013-04-22 20:23:15.382145000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTerrainLayerBlendMap.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreTerrainMaterialGenerator.h Main\include\auto\MogreTerrainMaterialGenerator.h
--- Main\include\auto\MogreTerrainMaterialGenerator.h	2013-04-22 20:21:41.616446900 +0200
+++ Main\include\auto\MogreTerrainMaterialGenerator.h	2013-04-22 20:23:15.454625900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTerrainMaterialGenerator.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 #include "MogrePixelFormat.h"
diff -r -u -N -w Main\include\auto\MogreTerrainMaterialGeneratorA.h Main\include\auto\MogreTerrainMaterialGeneratorA.h
--- Main\include\auto\MogreTerrainMaterialGeneratorA.h	2013-04-22 20:21:41.644466000 +0200
+++ Main\include\auto\MogreTerrainMaterialGeneratorA.h	2013-04-22 20:23:15.460629900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTerrainMaterialGeneratorA.h"
+#pragma managed(pop)
 #include "MogreTerrainMaterialGenerator.h"
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
diff -r -u -N -w Main\include\auto\MogreTerrainPagedWorldSection.h Main\include\auto\MogreTerrainPagedWorldSection.h
--- Main\include\auto\MogreTerrainPagedWorldSection.h	2013-04-22 20:21:41.658477000 +0200
+++ Main\include\auto\MogreTerrainPagedWorldSection.h	2013-04-22 20:23:15.468639300 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTerrainPagedWorldSection.h"
+#pragma managed(pop)
 #include "MogrePagedWorldSection.h"
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
diff -r -u -N -w Main\include\auto\MogreTerrainPaging.h Main\include\auto\MogreTerrainPaging.h
--- Main\include\auto\MogreTerrainPaging.h	2013-04-22 20:21:41.666480700 +0200
+++ Main\include\auto\MogreTerrainPaging.h	2013-04-22 20:23:15.476640200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTerrainPaging.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogrePlatform.h"
 
diff -r -u -N -w Main\include\auto\MogreTerrainQuadTreeNode.h Main\include\auto\MogreTerrainQuadTreeNode.h
--- Main\include\auto\MogreTerrainQuadTreeNode.h	2013-04-22 20:21:41.693500600 +0200
+++ Main\include\auto\MogreTerrainQuadTreeNode.h	2013-04-22 20:23:15.485650100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTerrainQuadTreeNode.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreTextAreaOverlayElement.h Main\include\auto\MogreTextAreaOverlayElement.h
--- Main\include\auto\MogreTextAreaOverlayElement.h	2013-04-22 20:21:41.714815400 +0200
+++ Main\include\auto\MogreTextAreaOverlayElement.h	2013-04-22 20:23:15.492656600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTextAreaOverlayElement.h"
+#pragma managed(pop)
 #include "MogreOverlayElement.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreTexture.h Main\include\auto\MogreTexture.h
--- Main\include\auto\MogreTexture.h	2013-04-22 20:21:41.747837600 +0200
+++ Main\include\auto\MogreTexture.h	2013-04-22 20:23:15.500659100 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTexture.h"
+#pragma managed(pop)
 #include "MogreResource.h"
 #include "MogrePrerequisites.h"
 #include "MogrePixelFormat.h"
diff -r -u -N -w Main\include\auto\MogreTextureManager.h Main\include\auto\MogreTextureManager.h
--- Main\include\auto\MogreTextureManager.h	2013-04-22 20:21:41.780859100 +0200
+++ Main\include\auto\MogreTextureManager.h	2013-04-22 20:23:15.509664500 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTextureManager.h"
+#pragma managed(pop)
 #include "MogreResourceManager.h"
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
diff -r -u -N -w Main\include\auto\MogreTextureUnitState.h Main\include\auto\MogreTextureUnitState.h
--- Main\include\auto\MogreTextureUnitState.h	2013-04-22 20:21:41.830892500 +0200
+++ Main\include\auto\MogreTextureUnitState.h	2013-04-22 20:23:15.521674600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreTextureUnitState.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreCommon.h"
 #include "MogreBlendMode.h"
diff -r -u -N -w Main\include\auto\MogreUnifiedHighLevelGpuProgram.h Main\include\auto\MogreUnifiedHighLevelGpuProgram.h
--- Main\include\auto\MogreUnifiedHighLevelGpuProgram.h	2013-04-22 20:21:41.859910100 +0200
+++ Main\include\auto\MogreUnifiedHighLevelGpuProgram.h	2013-04-22 20:23:15.527674900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreUnifiedHighLevelGpuProgram.h"
+#pragma managed(pop)
 #include "MogreHighLevelGpuProgram.h"
 #include "MogrePrerequisites.h"
 #include "MogreResource.h"
diff -r -u -N -w Main\include\auto\MogreUserObjectBindings.h Main\include\auto\MogreUserObjectBindings.h
--- Main\include\auto\MogreUserObjectBindings.h	2013-04-22 20:21:41.868920800 +0200
+++ Main\include\auto\MogreUserObjectBindings.h	2013-04-22 20:23:15.539684600 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreUserObjectBindings.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 
 namespace Mogre
diff -r -u -N -w Main\include\auto\MogreVertexBoneAssignment.h Main\include\auto\MogreVertexBoneAssignment.h
--- Main\include\auto\MogreVertexBoneAssignment.h	2013-04-22 20:21:41.872921800 +0200
+++ Main\include\auto\MogreVertexBoneAssignment.h	2013-04-22 20:23:15.548691200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreVertexBoneAssignment.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreVertexIndexData.h Main\include\auto\MogreVertexIndexData.h
--- Main\include\auto\MogreVertexIndexData.h	2013-04-22 20:21:41.883926500 +0200
+++ Main\include\auto\MogreVertexIndexData.h	2013-04-22 20:23:15.554694000 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreVertexIndexData.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreHardwareVertexBuffer.h"
 #include "MogreHardwareBuffer.h"
diff -r -u -N -w Main\include\auto\MogreViewport.h Main\include\auto\MogreViewport.h
--- Main\include\auto\MogreViewport.h	2013-04-22 20:21:41.905942100 +0200
+++ Main\include\auto\MogreViewport.h	2013-04-22 20:23:15.563701900 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreViewport.h"
+#pragma managed(pop)
 #include "MogrePrerequisites.h"
 #include "MogreFrustum.h"
 #include "MogrePlatform.h"
diff -r -u -N -w Main\include\auto\MogreWindowEventUtilities.h Main\include\auto\MogreWindowEventUtilities.h
--- Main\include\auto\MogreWindowEventUtilities.h	2013-04-22 20:21:41.918257200 +0200
+++ Main\include\auto\MogreWindowEventUtilities.h	2013-04-22 20:23:15.575707400 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreWindowEventUtilities.h"
+#pragma managed(pop)
 
 namespace Mogre
 {
diff -r -u -N -w Main\include\auto\MogreWireBoundingBox.h Main\include\auto\MogreWireBoundingBox.h
--- Main\include\auto\MogreWireBoundingBox.h	2013-04-22 20:21:41.928265900 +0200
+++ Main\include\auto\MogreWireBoundingBox.h	2013-04-22 20:23:15.584713200 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreWireBoundingBox.h"
+#pragma managed(pop)
 #include "MogreSimpleRenderable.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreWorkQueue.h Main\include\auto\MogreWorkQueue.h
--- Main\include\auto\MogreWorkQueue.h	2013-04-22 20:21:41.943274900 +0200
+++ Main\include\auto\MogreWorkQueue.h	2013-04-22 20:23:15.595722800 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreWorkQueue.h"
+#pragma managed(pop)
 #include "MogrePlatform.h"
 #include "MogrePrerequisites.h"
 
diff -r -u -N -w Main\include\auto\MogreZip.h Main\include\auto\MogreZip.h
--- Main\include\auto\MogreZip.h	2013-04-22 20:21:41.961291500 +0200
+++ Main\include\auto\MogreZip.h	2013-04-22 20:23:15.603725700 +0200
@@ -10,7 +10,9 @@
 
 #pragma once
 
+#pragma managed(push, off)
 #include "OgreZip.h"
+#pragma managed(pop)
 #include "MogreArchive.h"
 #include "MogrePrerequisites.h"
 
