diff -r -u -N -w Main\include\auto\MogreRoot.h Main\include\auto\MogreRoot.h
--- Main\include\auto\MogreRoot.h	2013-04-22 20:21:40.643423600 +0200
+++ Main\include\auto\MogreRoot.h	2013-04-22 20:29:22.258816200 +0200
@@ -60,12 +60,9 @@
     {
     }
   
+    // Destructor - this runs in the main thread -- if the user calls Dispose()
     ~Root()
     {
-      this->!Root();
-    }
-    !Root()
-    {
       _native = Ogre::Root::getSingletonPtr();
       
       // Collect all SharedPtr objects that are waiting for finalization
@@ -82,6 +79,10 @@
       _singleton = nullptr;
     }
   
+    // Finalizer - this runs in the garbage collector thread.
+    !Root()
+    {
+    }
   
     //Public Declarations
   public:
